go/private/context.bzl

```diff
@@ -267,13 +270,17 @@ def _go_context_data(ctx):
   linker_options = [o for o in raw_linker_options if not o in [
     "-Wl,--gc-sections",
   ]]
+  env = {}
+  tags = []
```

Comment:
I think the best way to populate this is by reading `ctx.var["gotags"]` or something similar. Users could set tags with `--define=gotags=foo,bar,baz` on the command line or in an rc file.

Reply:
this could be added later, but the main case for me to use it since the builders can too.
also, when building for ios, both tags and env need to be properly set up, and in that case I'm doing it from inside the rules themselves

Reply:
actually `ctx.var["gotags"] or []` is rather easy to do, i'm sold